packages/@lwc/ssr-compiler/src/compile-template/transformers/element.ts

```diff
@@ -246,6 +246,13 @@ export const Element: Transformer<IrElement | IrExternalComponent | IrSlot> = fu
         childContent = [];
     }
 
+    const isForeignElement = node.namespace !== HTML_NAMESPACE;
+    const hasChildren = childContent.length > 0;
+
+    if (isForeignElement && !hasChildren) {
```

Comment:
Are there any non-HTML elements that are _not_ self-closing when empty?

Reply:
I don't think so, or at least I couldn't find any SVG or MathML cases. The [spec](https://html.spec.whatwg.org/multipage/syntax.html#foreign-elements) just says "[Foreign elements](https://html.spec.whatwg.org/multipage/syntax.html#foreign-elements) must either have a start tag and an end tag, or a start tag that is marked as self-closing, in which case they must not have an end tag." but that is not exactly what you are asking.